adapters/medianet/medianet.go

```diff
@@ -0,0 +1,127 @@
+package medianet
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+
+	for _, imp := range request.Imp {
+		if err := preprocess(&imp); err != nil {
```

Comment:
Thank you for addressing comments. Code looks good to me. The only comment - do you really need this `preprocess` function at all? All it does - it unmarshalls imp ext and checks there are no errors. This part is covered by imp validator already. Please correct me if I miss something.

Reply:
Yes, you are right, earlier I added this function for unit test cases (for better coverage) in which we directly can Bidder. In future iterations, we have some plans to use that function in a much better way.

Reply:
UPDATE: I have removed `preprocess`